tests/core/system-settings/task.yaml

```diff
@@ -0,0 +1,12 @@
+summary: System settings work
+
+prepare: |
+    cp -aL /etc/hostname /tmp/hostname.bak
+
+restore: |
+    cp -a /tmp/hostname.bak /etc/hostname
+
+execute: |
+    echo "Check that setting the hostname works"
+    echo coole-kiste > /etc/hostname
```

Comment:
shouldn't we also check that the `hostname` command works and matches too?

Reply:
The `hostname` command doesn't care about the configuration file, it just sets and reads the hostname to/from the kernel, using  `sethostname(2)` and `gethostname(2)` (we would be testing the syscalls in this case -- we can do it, but it's orthogonal to the writable filesystem being mounted or not). IIRC in the old days we initialized the hostname using something like `hostname $(cat /etc/hostname)`, but currently it's probably a systemd thing.

Reply:
We should probably try `hostnamectl set-hostname` too.

Reply:
This is also indirectly tested via `tests/main/hostname-control`.